design-proposals/persist-vmi-firmware-uuid.md

```diff
@@ -0,0 +1,91 @@
+# Overview
+This proposal introduces a mechanism to persist the firmware UUID of a Virtual Machine Instance (VMI) in KubeVirt. By storing the firmware UUID, we ensure that it remains consistent across VMI restarts, which is crucial for applications and services that rely on the UUID for identification or licensing purposes.
+
+## Motivation
+In the current implementation, the firmware UUID of a Virtual Machine Instance (VMI) is generated dynamically during the creation of the VMI. This means that if the generation logic is changed in the future, the VMI may be assigned a different firmware UUID. This behavior can cause issues for applications that depend on a stable UUID.
+Furthermore, we plan to enhance the firmware UUID generation logic to include the VM's Name and Namespace, ensuring uniqueness across different namespaces. However, changing the generation logic could inadvertently assign new UUIDs to existing VMs upon restart, leading to potential disruptions.
+By persisting the firmware UUID in the VM's status, we can maintain consistency across restarts and future-proof existing VMs against changes in the UUID generation logic.
```

Comment:
I think that the motivation is simpler and starker. 

```suggestion
By definition, UUID must be a universally-unique identifier. The current implementation for the automatically-computed Firmware UUID is not universally unique: Two VMs with the same name get the same UUID, even if they reside in different namespace or different clusters. This proposal discusses how to fix this, while ensuring that already-identified VMs keep their current ID.
```

Reply:
fixed